src/matching.rs

```diff
@@ -0,0 +1,585 @@
+use std::collections::VecDeque;
+use std::hash::Hash;
+
+use super::visit::{
+    EdgeRef, GraphBase, IntoEdges, IntoNeighbors, IntoNodeIdentifiers, NodeCompactIndexable,
+    NodeIndexable, VisitMap, Visitable,
+};
+
+/// Computed
+/// [*matching*](https://en.wikipedia.org/wiki/Matching_(graph_theory)#Definitions)
+/// of the graph.
+pub struct Matching<G: GraphBase> {
+    graph: G,
+    mate: Vec<G::NodeId>,
+    n_edges: usize,
+}
+
+impl<G> Matching<G>
+where
+    G: NodeCompactIndexable,
```

Comment:
Not really, I need a map from node index to things. I used compact indexing to be able to use `Vec` instead of `HashMap` for performance reasons. But I realize this argument could be made in many algorithms in petgraph, yet possibility to use the algorithms with `StableGraph` is higher priority, which I understand.

Reply:
`NodeIndexable` has `node_bound`, so you can still use a `Vec` for `mate`. It will have unused holes, but that's similar to the implementation of `StableGraph`.

Reply:
Thanks, I will do that. For some reason I thought that it would be unacceptable, but didn't consider that it is how it is in `StableGraph` as well.